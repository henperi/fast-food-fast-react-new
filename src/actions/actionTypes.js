export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SET_AUTH_USER = 'SET_AUTH_USER';
export const REMOVE_AUTH_USER = 'REMOVE_AUTH_USER';
export const SET_ERRORS = 'SET_ERRORS';
export const GET_MENU = 'GET_MENU';
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const UPDATE_CART_ITEM_QUANTITY = 'UPDATE_CART_ITEM_QUANTITY';
export const TOGGLE_CART_SLIDER = 'TOGGLE_CART_SLIDER';
export const FETCH_CART_ITEMS = 'FETCH_CART_ITEMS';
export const SET_VISIBLE_CART = 'SET_VISIBLE_CART';
