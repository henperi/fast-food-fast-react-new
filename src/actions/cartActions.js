import {
  ADD_TO_CART, REMOVE_FROM_CART, TOGGLE_CART_SLIDER, SET_CURRENT_CART,
} from './actionTypes';

export const setCurrentCart = payload => ({
  type: SET_CURRENT_CART,
  payload,
});

export const addToCart = (id, name, price) => ({
  type: ADD_TO_CART,
  id,
  name,
  price,
});

export const removeFromCart = id => ({ type: REMOVE_FROM_CART, id });

export const updateCartQty = (id, updates) => ({
  type: 'UPDATE_CART_ITEM_QUANTITY',
  updates,
  id,
});

export const emptyCart = () => {
  localStorage.removeItem('foodItems');
  return { type: 'EMPTY_CART' };
};

export const toggleCartSlider = () => ({ type: TOGGLE_CART_SLIDER });
