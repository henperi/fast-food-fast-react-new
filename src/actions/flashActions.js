import { SHOW_FLASH, REMOVE_FLASH } from './actionTypes';

export const showFlash = (data = {}) => {
  const { title = '', message = '', flashType = 'success' } = data;
  return {
    type: SHOW_FLASH,
    payload: {
      title,
      message,
      flashType,
    },
  };
};

export const removeFlash = () => ({
  type: REMOVE_FLASH,
});
