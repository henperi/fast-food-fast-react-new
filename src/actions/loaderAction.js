import { TOGGLE_LOADER } from './actionTypes';

const toggleLoader = () => ({
  type: TOGGLE_LOADER,
});

export default toggleLoader;
