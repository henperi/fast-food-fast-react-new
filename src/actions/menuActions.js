import { SET_MENU, SET_ERRORS, SET_FEATURED_MENU } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';

export const setMenu = (payload = {}) => ({
  type: SET_MENU,
  payload,
});

export const setFeaturedMenu = (payload = {}) => {
  const featuredMenu = payload.slice(0, 3);

  return {
    type: SET_FEATURED_MENU,
    payload: featuredMenu,
  };
};

export const getMenu = () => (dispatch) => {
  axiosInstance
    .get('/menu')
    .then((response) => {
      const { data } = response;
      dispatch({
        type: SET_MENU,
        payload: { menu: [...data.menu] },
      });
      dispatch(setFeaturedMenu(data.menu));
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const updateMenu = () => {};
