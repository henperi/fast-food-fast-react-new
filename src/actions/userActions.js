import axiosInstance, { setAxiosToken } from '../utils/axiosInstance';
import { SET_ERRORS, SET_AUTH_USER, REMOVE_AUTH_USER } from './actionTypes';

export const setAuthUser = (payload = {}) => ({
  type: SET_AUTH_USER,
  payload,
});

export const logoutUser = () => ({
  type: REMOVE_AUTH_USER,
  payload: {
    isAuthenticated: false,
  },
});

const signupUrl = '/auth/signup';
const loginUrl = '/auth/login';

export const signupUser = (userData = {}, history) => (dispatch) => {
  axiosInstance
    .post(signupUrl, userData)
    .then((response) => {
      const { data } = response;
      const isAuthenticated = true;
      const { userToken } = data;
      const { userId, role } = data.createdUser;
      const name = data.createdUser.fullname;

      localStorage.setItem('userToken', userToken);
      localStorage.setItem('userId', userId);
      localStorage.setItem('fullname', name);
      localStorage.setItem('role', role);

      setAxiosToken(userToken);

      dispatch({
        type: SET_AUTH_USER,
        payload: {
          isAuthenticated,
          userId,
          userToken,
          role,
          fullname: name,
        },
      });

      return history.push('/users/foods');
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      // console.log(response);
      // console.log('=======');
      // console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const loginUser = (userData = {}, history) => (dispatch) => {
  axiosInstance
    .post(loginUrl, userData)
    .then((response) => {
      // console.log('=============', response, '=============');
      const { data } = response;
      const isAuthenticated = true;
      const { userToken } = data;
      const { fullname, role, userId } = data.userData;

      localStorage.setItem('userToken', userToken);
      localStorage.setItem('fullname', fullname);
      localStorage.setItem('role', role);
      localStorage.setItem('userId', userId);

      setAxiosToken(userToken);

      dispatch(
        setAuthUser({
          isAuthenticated,
          fullname,
          role,
          userId,
          userToken,
        }),
      );

      return history.push('/users/foods');
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      // console.log(response);
      // console.log('=======');
      // console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};
