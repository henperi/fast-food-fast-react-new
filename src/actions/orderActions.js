import { SET_ERRORS } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';
import { emptyCart } from './cartActions';

const placeOrder = (cartItems = {}, handleModal) => (dispatch) => {
  console.log(cartItems);
  const orderData = {
    foodItems: cartItems,
  };
  axiosInstance
    .post('/orders', orderData)
    .then((response) => {
      const { data } = response;
      console.log(data);
      handleModal();
      dispatch(emptyCart());
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(errors);
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export default placeOrder;
