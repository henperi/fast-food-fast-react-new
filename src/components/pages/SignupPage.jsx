import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { signupUser } from '../../actions/userActions';

class SignupPage extends Component {
  state = {
    email: '',
    password: '',
    passwordConfirmation: '',
    fullname: '',
    mobile: '',
    address: '',
    errors: [],
    isLoading: false,
  };

  componentWillMount() {
    this.props.user.isAuthenticated && this.props.history.push('/users/foods');
  }

  componentWillReceiveProps(nextProps) {
    return nextProps.errors && this.setState({ errors: nextProps.errors });
  }

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.setState({ isLoading: true });

    const {
      email, password, passwordConfirmation, fullname, mobile, address,
    } = this.state;

    const signupData = {
      email,
      password,
      passwordConfirmation,
      fullname,
      mobile,
      address,
    };

    const { signupUser: dispatchSignup, history } = this.props;

    dispatchSignup(signupData, history);
  };

  render() {
    const {
      email,
      password,
      passwordConfirmation,
      fullname,
      mobile,
      address,
      errors,
      isLoading,
    } = this.state;

    return (
      <div className="container">
        <div className="text-center contain-50">
          <div className="card card-shadow text-inverse">
            <h2>Create An Account To Get Started</h2>
            <p>It only takes a few seconds to signup, Lets get started!</p>

            <form className="form-data contain" onSubmit={this.onSubmit}>
              <div className="form-member text-left">
                <span>Email</span>
                <input
                  type="email"
                  name="email"
                  value={email}
                  className="form-input email"
                  placeholder="Enter Your Email"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'email' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member text-left">
                <span>Password</span>
                <input
                  type="password"
                  name="password"
                  value={password}
                  className="form-input password"
                  placeholder="Enter your password"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'password' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member text-left">
                <span>Repeat Password</span>
                <input
                  type="password"
                  name="passwordConfirmation"
                  value={passwordConfirmation}
                  className="form-input passwordConfirmation"
                  placeholder="Repeat your password"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'passwordConfirmation' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member text-left">
                <span>Fullname</span>
                <input
                  type="text"
                  name="fullname"
                  value={fullname}
                  className="form-input fullname"
                  placeholder="Enter Your Full Name"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'fullname' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member text-left">
                <span>Mobile</span>
                <input
                  type="tel"
                  name="mobile"
                  value={mobile}
                  className="form-input mobile"
                  placeholder="Enter Your Mobile"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'mobile' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member text-left">
                <span>Address</span>
                <input
                  type="text"
                  name="address"
                  value={address}
                  className="form-input address"
                  placeholder="Enter Your Home Address"
                  onChange={this.onChange}
                />
                {errors.map(
                  error => error.param === 'address' && (
                  <li key={`${error.param}-${Math.random()}`} className="list-item error">
                    {error.msg}
                  </li>
                  ),
                )}
              </div>
              <div className="form-member">
                <div className="response-area">
                  {errors.length === 1 && errors[0].msg && (
                    <li key={`${errors.length}-${Math.random()}`} className="list-item error">
                      {errors[0].msg}
                    </li>
                  )}
                </div>
                <div className={`loader ${isLoading ? '' : 'hide'}`}>
                  <i className="fa fa-spinner fa-spin" />
                </div>
                <button type="submit" name="signup" className="btn btn-blue btn-block signup">
                  Signup
                </button>
              </div>
              <p>
                Already have an account?
                <Link to="/signin" className="link">
                  Signin Instead
                </Link>
              </p>
            </form>
            <p>
              Go back home
              <Link to="/" className="link">
                Home
              </Link>
            </p>
          </div>
        </div>
      </div>
    );
  }
}

SignupPage.propTypes = {
  signupUser: PropTypes.func.isRequired,
  errors: PropTypes.array,
  history: PropTypes.object,
};

SignupPage.defaultProps = {
  errors: [],
};

const mapStateToProps = state => ({
  user: state.user,
  errors: state.errors,
});

export default connect(
  mapStateToProps,
  { signupUser },
)(withRouter(SignupPage));
