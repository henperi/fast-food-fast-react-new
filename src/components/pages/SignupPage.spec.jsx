/* eslint-disable no-undef */
import React from 'react';
import sinon from 'sinon';
import { SignupPage } from './SignupPage';
import { getOrderHistory } from '../../actions/orderActions';
import { signupUser } from '../../actions/userActions';
// import { loginUser } from '../../actions/userActions';
// import SignupForm from './SignupForm';

describe('The SignupPage Component Test Suite', () => {
  const wrapper = shallow(<SignupPage match={{ params: {} }} user={{}} signupUser={signupUser} />);
  it('should render the <OrderHistory /> Component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(SignupPage)).toBeDefined();
  });

  it('should handle input change event for username', () => {
    const event = { target: { name: 'email', value: 'henry' }, preventDefault: () => {} };

    wrapper.instance().onChange(event);
    expect(wrapper.state().email).toEqual('henry');
    expect(wrapper.instance().state.email).toEqual(event.target.value);
  });

  it('should call the handleAddToCart', () => {
    const event = { preventDefault: () => {} };

    wrapper.instance().onSubmit(event);
    expect(sinon.stub(jest.fn(wrapper.instance().props.signupUser()))).not.toHaveBeenCalled();
    // expect(
    //   sinon.stub(jest.fn(wrapper.instance().props.actions(reportArticle()))),
    // ).toHaveBeenCalled();
    // expect(wrapper.instance().state.reportType).toEqual(event.target.value);
  });
});
