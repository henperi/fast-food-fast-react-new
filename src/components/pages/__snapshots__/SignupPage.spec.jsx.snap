// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignupPage Component Test Suite should render the <OrderHistory /> Component 1`] = `
<div
  className="container"
>
  <div
    className="text-center contain-50"
  >
    <div
      className="card card-shadow text-inverse"
    >
      <h2>
        Create An Account To Get Started
      </h2>
      <p>
        It only takes a few seconds to signup, Lets get started!
      </p>
      <form
        className="form-data contain"
        onSubmit={[Function]}
      >
        <div
          className="form-member text-left"
        >
          <span>
            Email
          </span>
          <input
            className="form-input email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-member text-left"
        >
          <span>
            Password
          </span>
          <input
            className="form-input password"
            name="password"
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-member text-left"
        >
          <span>
            Repeat Password
          </span>
          <input
            className="form-input passwordConfirmation"
            name="passwordConfirmation"
            onChange={[Function]}
            placeholder="Repeat your password"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-member text-left"
        >
          <span>
            Fullname
          </span>
          <input
            className="form-input fullname"
            name="fullname"
            onChange={[Function]}
            placeholder="Enter Your Full Name"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-member text-left"
        >
          <span>
            Mobile
          </span>
          <input
            className="form-input mobile"
            name="mobile"
            onChange={[Function]}
            placeholder="Enter Your Mobile"
            type="tel"
            value=""
          />
        </div>
        <div
          className="form-member text-left"
        >
          <span>
            Address
          </span>
          <input
            className="form-input address"
            name="address"
            onChange={[Function]}
            placeholder="Enter Your Home Address"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-member"
        >
          <div
            className="response-area"
          />
          <div
            className="loader hide"
          >
            <i
              className="fa fa-spinner fa-spin"
            />
          </div>
          <button
            className="btn btn-blue btn-block signup"
            name="signup"
            type="submit"
          >
            Signup
          </button>
        </div>
        <p>
          Already have an account?
          <Link
            className="link"
            replace={false}
            to="/signin"
          >
            Signin Instead
          </Link>
        </p>
      </form>
      <p>
        Go back home
        <Link
          className="link"
          replace={false}
          to="/"
        >
          Home
        </Link>
      </p>
    </div>
  </div>
</div>
`;
